src/context.ts

```diff
@@ -90,14 +98,15 @@ export let getBuildCache: (ctx: FreshContext<unknown>) => BuildCache;
 
 export class FreshReqContext<State>
   implements FreshContext<State>, PageProps<unknown, State> {
-  config: ResolvedFreshConfig;
-  url: URL;
-  req: Request;
-  params: Record<string, string>;
-  state: State = {} as State;
+  readonly config: ResolvedFreshConfig;
+  readonly url: URL;
+  readonly req: Request;
+  readonly params: Record<string, string>;
+  readonly state: State = {} as State;
   data: unknown = undefined;
   error: unknown | null = null;
-  info: Deno.ServeHandlerInfo | Deno.ServeHandlerInfo;
+  readonly info: Deno.ServeHandlerInfo | Deno.ServeHandlerInfo;
+  readonly isPartial: boolean;
```

Comment:
Most of the properties could safely add `readonly` here to match the interface without TS errors. But I can revert this if preferred, but I think it can help minimize mistakes.